### PR TITLE
Feature/icky 1063

### PR DESCRIPTION
**Pull request by kibo-sushant**

### CODE REVIEW

File: components/instant-delivery/DeliveryAddress/DeliveryAddress.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  storeBoundary,
+  setStoreBoundary,
+  deliveryAddress,
+  setDeliveryAddress,
+}: Props) => {
+  const { t } = useTranslation('common')
+  const addressSchema = useFormSchema()
+
+  const { data: newStoreBoundary } = useGetStoreServiceBoundary(deliveryAddress)
+  setStoreBoundary(newStoreBoundary)
```
Comment (kibo-chandradeeptalaha) [inline components/instant-delivery/DeliveryAddress/DeliveryAddress.tsx:40]: Won't this cause multiple calls ? Also, till the time newStoreBoundary is not available, it'll still get called right ?

File: components/instant-delivery/InstantDeliveryTemplate/InstantDeliveryTemplate.tsx [hunk 2]
```diff
@@ -1,6 +1,102 @@
...
 
-const InstantDeliveryTemplate = (props: any) => {
-  return <h1>Instant Delivery Template</h1>
+import { Box, Stack, Step, Stepper, Typography, Slider, StepButton } from '@mui/material'
+import { useTranslation } from 'next-i18next'
+
+import DeliveryAddress from '@/components/instant-delivery/DeliveryAddress/DeliveryAddress'
+import { DeliveryLocation } from '@/hooks'
+
+const stepperStyles = {
```
Comment (kibo-chandradeeptalaha) [inline components/instant-delivery/InstantDeliveryTemplate/InstantDeliveryTemplate.tsx:9]: try to put the styles in a separate style file

File: components/instant-delivery/InstantDeliveryTemplate/InstantDeliveryTemplate.tsx [hunk 3]
```diff
@@ -1,6 +1,102 @@
...
+        <div>
+          <DeliveryAddress
+            storeBoundary={storeBoundary}
+            setStoreBoundary={setStoreBoundary}
+            deliveryAddress={deliveryAddress}
+            setDeliveryAddress={setDeliveryAddress}
+          />
+        </div>
+        <div>
+          <h1>Delivery Window </h1>
```
Comment (kibo-chandradeeptalaha) [inline components/instant-delivery/InstantDeliveryTemplate/InstantDeliveryTemplate.tsx:94]: use localizations everywhere if applicable

File: lib/api/util/get-paypal-checkout-settings.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { apiAuthClient } from './api-auth-client'
+
+type CheckoutSettingsResponse = {
+  userName: string
+  password: string
+  orderProcessing: string
+}
+const getCheckoutSettings = (res: any): CheckoutSettingsResponse => {
+  const payPalExpress2 = res.paymentSettings.externalPaymentWorkflowDefinitions.find(
+    (item: any) => item.name === 'PayPalExpress2'
```
Comment (kibo-chandradeeptalaha) [inline lib/api/util/get-paypal-checkout-settings.ts]: use the enum for 'PayPalExpress2'

File: lib/helpers/buildPayPalPaymentActionForCheckoutParams.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  currencyCode: string,
+  checkout: CrOrder | Checkout,
+  billingAddress: CrContact | undefined,
+  externalTransactionId: string,
+  payerId: string
+): PaymentActionInput => {
+  const billingInfo: CrBillingInfo = {
+    billingContact: { ...billingAddress, email: billingAddress?.email || checkout?.email },
+    card: null,
+    paymentType: 'PayPalExpress2',
```
Comment (kibo-chandradeeptalaha) [inline lib/helpers/buildPayPalPaymentActionForCheckoutParams.ts]: use enum for 'PayPalExpress2'

File: hooks/custom/usePayPalBearerToken/usePayPalBearerToken.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        const res = await fetch(url, {
+          headers: {
+            Accept: 'application/json',
+          },
+        })
+        const data = await res.json()
+
+        setBearerToken(data)
+      } catch (error) {
+        console.error('Error:', error)
```
Comment (kibo-chandradeeptalaha) [inline hooks/custom/usePayPalBearerToken/usePayPalBearerToken.tsx]: use logger

File: components/instant-delivery/DeliveryAddress/DeliveryAddress.tsx [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    setStoreBoundary(undefined)
+    setDeliveryAddress({ ...data })
+  }
+
+  // clear storeBoundary when address is changed
+  useEffect(() => {
+    if (storeBoundary && isDirty) {
+      setStoreBoundary(undefined)
+      setDeliveryAddress(undefined)
+      console.log(`clear storeBoundary`)
```
Comment (kibo-chandradeeptalaha) [inline components/instant-delivery/DeliveryAddress/DeliveryAddress.tsx]: remove consoles

File: components/checkout/PaymentStep/PayPalButton.tsx [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+
+    const body = {
+      payment_source,
+    }
+
+    try {
+      await voidActivePayment()
+      await addPaymentMethod(orderID, payerID)
+    } catch (error) {
+      console.error(`Error while confirming paypal order: ${orderID}`, error)
```
Comment (kibo-chandradeeptalaha) [inline components/checkout/PaymentStep/PayPalButton.tsx]: use logger where applicable

**Review by kibo-chandradeeptalaha — APPROVED** [review]
Please check the comments and handle accordingly. Otherwise, looks good

File: __mocks__/stories/paypalCheckoutSettingsMock.ts [hunk 9]
```diff
@@ -0,0 +1,103 @@
+export const paypalCheckoutSettingsMock = {
+  paymentSettings: {
+    externalPaymentWorkflowDefinitions: [
```
Comment (Kibo-TriptiV) [inline __mocks__/stories/paypalCheckoutSettingsMock.ts]: why Paypal changes are part of delivery solution? can't be separate PR?

File: components/instant-delivery/InstantDeliveryTemplate/InstantDeliveryTemplate.tsx (hunk 3 above)
Comment (kibo-sushant) [reply components/instant-delivery/InstantDeliveryTemplate/InstantDeliveryTemplate.tsx:94]: This story is yet to develop

File: lib/api/util/get-paypal-checkout-settings.ts (hunk 4 above)
Comment (kibo-sushant) [reply lib/api/util/get-paypal-checkout-settings.ts]: It's not part of the story. Rebased the geetanshu's code.

File: lib/helpers/buildPayPalPaymentActionForCheckoutParams.ts (hunk 5 above)
Comment (kibo-sushant) [reply lib/helpers/buildPayPalPaymentActionForCheckoutParams.ts]: It's not part of the story. Rebased the geetanshu's code.

File: hooks/custom/usePayPalBearerToken/usePayPalBearerToken.tsx (hunk 6 above)
Comment (kibo-sushant) [reply hooks/custom/usePayPalBearerToken/usePayPalBearerToken.tsx]: It's not part of the story. Rebased the geetanshu's code.

File: components/checkout/PaymentStep/PayPalButton.tsx (hunk 8 above)
Comment (kibo-sushant) [reply components/checkout/PaymentStep/PayPalButton.tsx]: It's not part of the story. Rebased the geetanshu's code.

File: components/instant-delivery/DeliveryAddress/DeliveryAddress.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    expect(screen.getByText(/delivery-solutions-zipcode/i)).toBeVisible()
+
+    screen.getByRole('textbox', { name: /delivery-solutions-street/i })
+    screen.getByRole('textbox', { name: /delivery-solutions-city/i })
+    screen.getByRole('textbox', { name: /delivery-solutions-country/i })
+    screen.getByRole('textbox', { name: /delivery-solutions-state/i })
+    screen.getByRole('textbox', { name: /delivery-solutions-zipcode/i })
+
+    screen.getByRole('button', { name: /confirm\-address/i })
+  })
```
Comment (Kibo-TriptiV) [inline components/instant-delivery/DeliveryAddress/DeliveryAddress.spec.tsx:46]: textbox and button should be inside expect to visible or rendered on dom.

File: components/instant-delivery/DeliveryAddress/DeliveryAddress.tsx (hunk 1 above)
Comment (kibo-sushant) [reply components/instant-delivery/DeliveryAddress/DeliveryAddress.tsx:40]: There won't be multiple calls as we have handled it inside useGetStoreServiceBoundar (enabled: !!deliveryAddress,)

**Review by Kibo-TriptiV — APPROVED** [review]
please check minor comments.

File: components/instant-delivery/DeliveryAddress/DeliveryAddress.spec.tsx (hunk 10 above)
Comment (kibo-sushant) [reply components/instant-delivery/DeliveryAddress/DeliveryAddress.spec.tsx:46]: Rebased base repo created by Geetanshu

File: components/instant-delivery/DeliveryAddress/DeliveryAddress.tsx (hunk 7 above)
Comment (kibo-sushant) [reply components/instant-delivery/DeliveryAddress/DeliveryAddress.tsx]: Done

File: components/instant-delivery/InstantDeliveryTemplate/InstantDeliveryTemplate.tsx (hunk 2 above)
Comment (kibo-sushant) [reply components/instant-delivery/InstantDeliveryTemplate/InstantDeliveryTemplate.tsx:9]: There are just 7 props, I think it's fine
